### ISSUE TITLE
Create a complete and consice documentation for GDExtension

### ISSUE THREAD

**Issue by nonunknown** [godotengine/godot-proposals#3980]
### Describe the project you are working on

A G.O.A.P Implementation for Godotengine via GDExtension

### Describe the problem or limitation you are having in your project

Me and a lot of users has little to none knowledge about GDNative and C++, now that GDNative is "deprecated" in 4.x we have even less knowledge about such powerful tool, some of them:

* Classes we have access to
* Classes which are not exposed to API but we can use in C++ side

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

The documentation will have a "Zero to Hero" object, where a user with basic knowledge of GDExtension will learn:

* Getting the necessary files to work on
* What is each file in there does
* Generate the bindings (for each platform, and cross-platform)
* Creating the First Extension
* `Ref<>` and `*` how and when to use them
* Similarities with the core source code
   - How it works in core, and how to reproduce in GDExtension
   - How to deal with virtuals (overriding a C++ function via GDScript, like `_ready()`)
 * About Classes from GodotEngine core we can use in C++ but are not exposed in the API
 * Creating a simple game
 * Creating a plugin
 * Overriding/Extending Engine features (if possible)
 * Compiling your extensions
 * Advanced Stuff
    - Optimization
    - Debugging
    - File Structure
    - Many More
 

I may have forgot some stuff, so if you're reading this, please comment below suggestions

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

A Complete documentation in the docs page

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Nope, its a documentation.

### Is there a reason why this should be core and not an add-on in the asset library?

this cant be addon, since is GDExtension.

**Comment by Gnumaru** [comment]
Shouldn't this belong to the other project, godot docs?

https://github.com/godotengine/godot-docs/

**Comment by nonunknown** [comment]
not sure, there is for bugs and enhacements related to existing documentation, @Calinou can confirm?

**Comment by Xrayez** [comment]
GDExtension appears to be still work-in-progress. But GDExtension without even preliminary documentation/specification is going to be quite useless, by useless I mean this may prevent adoption by potential developers that would like to improve and build upon GDExtension (like me). I think GDExtension should get more love and support, so I'd leave this proposal here... 😛

I see no technical specification for how GDExtension should work or supposed to work, not even talking about user documentation.

See also proposal of mine:
- https://github.com/godotengine/godot-proposals/issues/3864

**Comment by nonunknown** [comment]
Yep GDExtension is WIP, which means the best time for thinking about how to present the documentation for end-users is now! Also I dont think GDExtension is "useless" it makes life easier when you need to deal with more advanced stuff, having the benefits of the front-end (godot editor) and back-end (core source code).
